web-ng/src/app/components/layer-list-item/layer-list-item.component.ts

```diff
@@ -139,13 +137,9 @@ export class LayerListItemComponent implements OnInit, OnDestroy {
     );
   }
 
-  initLayerItemPermission(): void {
-    const project = this.projectService.getCurrentProject();
-    if (!project) {
-      return;
-    }
-    const acl = this.projectService.getProjectAcl(project);
-    this.canCustomizeLayer = this.authService.canManageProject(acl);
+  canShare(): boolean {
+    const acl = this.projectService.getCurrentProjectAcl();
```

Comment:
Could you kindly let `getCurrentProjectAcl()` return `[]` instead of `undefined`? 

Also, we'll never need to call `canManageProject()` on an arbitrary ACL; it will always be on the current project, so you can move the call to `getCurrentProjectAcl()` into `canManageProject()`.

Then `canShare` will be just:

```
  return this.authService.canManageProject();
```

Wdyt?

Reply:
I moved the logic of canManageProject to the project service as it will result in circular dependency if I keep the logic in auth service.